target_chains/solana/programs/pyth-price-publisher/src/validate.rs

```diff
@@ -0,0 +1,122 @@
+use {
+    crate::{
+        accounts,
+        ensure,
+        instruction::{
+            CONFIG_SEED,
+            PUBLISHER_CONFIG_SEED,
+        },
+    },
+    solana_program::{
+        account_info::AccountInfo,
+        program_error::ProgramError,
+        program_memory::sol_memcmp,
+        pubkey::Pubkey,
+        system_program,
+    },
+};
+
+pub fn validate_publisher<'a, 'b>(
+    account: Option<&'b AccountInfo<'a>>,
+) -> Result<&'b AccountInfo<'a>, ProgramError> {
+    let publisher = account.ok_or(ProgramError::NotEnoughAccountKeys)?;
+    ensure!(ProgramError::MissingRequiredSignature, publisher.is_signer);
+    ensure!(ProgramError::InvalidArgument, publisher.is_writable);
+    Ok(publisher)
+}
+
+pub fn validate_system<'a, 'b>(
+    account: Option<&'b AccountInfo<'a>>,
+) -> Result<&'b AccountInfo<'a>, ProgramError> {
+    let system = account.ok_or(ProgramError::NotEnoughAccountKeys)?;
+    ensure!(
+        ProgramError::InvalidArgument,
+        pubkey_eq(system.key, &system_program::id())
+    );
+    Ok(system)
+}
+
+pub fn validate_payer<'a, 'b>(
+    account: Option<&'b AccountInfo<'a>>,
+) -> Result<&'b AccountInfo<'a>, ProgramError> {
+    let payer = account.ok_or(ProgramError::NotEnoughAccountKeys)?;
+    ensure!(ProgramError::MissingRequiredSignature, payer.is_signer);
+    ensure!(ProgramError::InvalidArgument, payer.is_writable);
+    Ok(payer)
+}
+
+fn pubkey_eq(a: &Pubkey, b: &Pubkey) -> bool {
```

Comment:
just for extra caution let's add `inline` here.

Reply:
Done.